### PR TITLE
Multi Group-by

### PR DESCRIPTION
**Pull request by aroc**
#### Note: Until Keen.js supports this, we cannot merge this PR.
This is now supported in Keen.js `3.4.0-rc2`. It will be a full release by the time we cut a new Explorer release - so this can be merged into master now.

### What does this PR do? How does it affect users?
It adds support in Explorer to optionally group by two properties. Our API supports this.

### How should this be tested (feature switches, URLs, special user permissions)?
Try using the new multi-group by feature in the Query Builder section and see if the UI works as expected.

### Related Tickets
None.

### CODE REVIEW

**Comment by joannecheng** [comment]
Minor nitpick: when I load a saved query with multiple group bys (or any group bys), the group by toggle doesn't open.

File: client/js/app/components/explorer/query_builder/group_by_field.js [hunk 1]
```diff
@@ -24,25 +24,79 @@ var GroupByField = React.createClass({
...
+  },
+
   // React methods
 
   render: function() {
+    var secondGroupByField;
+    var toggleText = "Group by a second property";
+    var toggleIcon = "plus";
+
+    if (this.props.value.length > 1) {
```
Comment (joannecheng) [inline client/js/app/components/explorer/query_builder/group_by_field.js]: how would you feel about pulling out the logic to generate the select fields and the multigroup toggle in a different private method?

File: client/js/app/components/explorer/query_builder/group_by_field.js [hunk 2]
```diff
@@ -24,25 +24,79 @@ var GroupByField = React.createClass({
     return this.props.value;
   },
 
+  handleChange: function(name, value) {
+    var newVal = this.props.value.slice();
+    var index = name.split('.')[1];
+    newVal[index] = value;
+    this.props.handleChange(name.split('.')[0], newVal);
```
Comment (josephwegner) [inline client/js/app/components/explorer/query_builder/group_by_field.js]: is `name.split('.')[0]` ever not `'group_by'`? Why not just hard code that?

File: client/js/app/components/explorer/query_builder/group_by_field.js [hunk 3]
```diff
@@ -24,25 +24,79 @@ var GroupByField = React.createClass({
...
+  },
+
+  toggleMultiGroupBy: function(event) {
+    event.preventDefault();
+    var newVal;
+    switch (this.props.value.length) {
+      case 0:
```
Comment (josephwegner) [inline client/js/app/components/explorer/query_builder/group_by_field.js]: when is this case hit?

File: client/js/app/components/explorer/query_builder/group_by_field.js (hunk 1 above)
Comment (aroc) [inline client/js/app/components/explorer/query_builder/group_by_field.js]: Yeah not a bad idea. Happy to do that! :)

File: client/js/app/components/explorer/query_builder/group_by_field.js (hunk 2 above)
Comment (aroc) [inline client/js/app/components/explorer/query_builder/group_by_field.js]: haha great eyes! You're absolutely correct here. Hard coding it is.

File: client/js/app/components/explorer/query_builder/group_by_field.js (hunk 3 above)
Comment (aroc) [inline client/js/app/components/explorer/query_builder/group_by_field.js]: The array is empty at first when no `group_by` has been set yet. So initial state basically.

**Comment by aroc** [comment]
@joannecheng Not a minor nitpick, that's a bug. I'll look into it.

**Comment by aroc** [comment]
Ok, this is ready for another review. @joannecheng Your issue with saved query loading has been fixed. The GroupByFields component has been refactored to make use of functions for some of the JSX to clean things up as well.

File: client/js/app/components/explorer/query_builder/group_by_field.js (hunk 3 above)
Comment (josephwegner) [inline client/js/app/components/explorer/query_builder/group_by_field.js]: +1 makes sense.

**Comment by josephwegner** [comment]
+1

**Comment by aroc** [comment]
Ok, this is ready for another quick review. @joannecheng The group by fields *do* open as expected now when loading from a saved query. :)

**Comment by joannecheng** [comment]
:+1: Works for me now!

**Comment by dustinlarimer** [comment]
:+1: all good here

**Comment by aroc** [comment]
Merging!

**Pull request merged**